rotkehlchen/serialization/deserialize.py

```diff
@@ -578,10 +607,36 @@ def deserialize_evm_transaction(
             gas_used = read_integer(data, 'gasUsed', source)
         nonce = read_integer(data, 'nonce', source)
 
-        if chain_id in L2_CHAINIDS_WITH_L1_FEES and evm_inquirer is not None:
-            if not raw_receipt_data:
-                raw_receipt_data = evm_inquirer.get_transaction_receipt(tx_hash)
-            l1_fee = maybe_read_integer(raw_receipt_data, 'l1Fee', source)
+        if chain_id in L2_CHAINIDS_WITH_L1_FEES:
+            if (
+                (l1_fee := data.get('L1FeesPaid')) is None and  # etherscan txlist uses this key
```

Comment:
you don't run any sanitation on `l1_fee := data.get('L1FeesPaid')`. What if it's a string and not an int? Other parts of the code here seem to check the contents.